ggshield/scan/scanner.py

```diff
@@ -175,7 +175,13 @@ def __init__(
         ignored_matches: Optional[Iterable[IgnoredMatch]] = None,
         ignored_detectors: Optional[Set[str]] = None,
         ignore_known_secrets: Optional[bool] = None,
+        check_api_health: Optional[bool] = True,
     ):
+        if check_api_health:
+            health_response = client.health_check()
+            if not health_response.success:
+                raise click.ClickException(health_response.detail)
```

Comment:
Two things here:

1. You can turn this code into a static method like `check_can_scan()`. This way you can reuse it in `repo.scan_commit_range()`
2. The exception should be AuthError when health_response.status_code is 401 and UnexpectedError otherwise.